staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
+// if gmf.GroupArgs.GroupName is already registered in groupFactoryRegistry,
+// instead of panicing.
+func (gmf *GroupMetaFactory) AnnounceIfNotAnnounced(groupFactoryRegistry APIGroupFactoryRegistry) *GroupMetaFactory {
+	if _, ok := groupFactoryRegistry[gmf.GroupArgs.GroupName]; ok {
```

Comment:
I totally agree after we move the install calls next to Scheme+Registry, i.e. same file. We had a similar situation with registered types, i.e. registering same type with multiple GVs. In theory it's predictable, in practice we had two handful of inconcistencies.